lib/bing_translator.rb

```diff
@@ -16,18 +16,21 @@ class BingTranslator
   NAMESPACE_URI = 'http://api.microsofttranslator.com/V2'
   ACCESS_TOKEN_URI = 'https://datamarket.accesscontrol.windows.net/v2/OAuth2-13'
   DATASETS_URI = "https://api.datamarket.azure.com/Services/My/Datasets?$format=json"
+  COGNITIVE_ACCESS_TOKEN_URI = 'https://api.cognitive.microsoft.com/sts/v1.0/issueToken'
 
   class Exception < StandardError; end
   class AuthenticationException < StandardError; end
 
-  def initialize(client_id, client_secret, skip_ssl_verify = false, account_key = nil)
+  def initialize(client_id: nil, client_secret: nil, skip_ssl_verify: false, account_key: nil, subscription_key: nil)
```

Comment:
keep in mind, that keyword arguments are only available in ruby >= 2.0, so support for 1.9.3 is dropped.
Maybe create a separate branch/release for azure support?